File: app/DLCosmicTag/tests/run_test.py
from __future__ import print_function
import sys,os
import ROOT as rt
from larlite import larlite
from larcv import larcv

# Test function opens larcv and larlite file


larcv_input   = sys.argv[1]
larlite_input = sys.argv[2]

iolarcv = larcv.IOManager( larcv.IOManager.kREAD )
iolarcv.add_in_file( larcv_input )
iolarcv.initialize()

out = larcv.IOManager( larcv.IOManager.kWRITE )
out.set_out_file( "out_test.root" )
out.initialize()

cfg  = larcv.CreatePSetFromFile( "test_dlcosmicutil.fcl", "test" )
pset = cfg.get_pset("TestDLCosmicUtil")
print ("--- PSET ---------------")
print (pset.dump())
print ("------------------------")

algo = larcv.DLCosmicTagUtil()
algo.Configure(pset)
algo.set_verbosity(0)

nentries = iolarcv.get_n_entries()

logger = larcv.logger.get_shared()

for ientry in xrange(nentries):

    msg = "Entry {}\n".format(ientry)
    logger.send(2,"Event Loop").write( msg, len(msg) )
    
    iolarcv.read_entry(ientry)
    algo.go_to_entry(ientry)

    

    # whole adc larcv image_v
    event_img_v  = iolarcv.get_data( larcv.kProductImage2D, "wire" )
    img_v = event_img_v.Image2DArray()

    masked = algo.makeCosmicMaskedImage( img_v )

    # output images
    evout_wire   = out.get_data( larcv.kProductImage2D, "wire" )
    evout_masked = out.get_data( larcv.kProductImage2D, "intimemask" )

    for iimg in xrange( img_v.size() ):
        evout_wire.Append(   img_v.at(iimg) )
        evout_masked.Append( masked.at(iimg) )

    out.set_id( iolarcv.event_id().run(), iolarcv.event_id().subrun(), iolarcv.event_id().event() )
    out.save_entry()


out.finalize()
iolarcv.finalize()
print ("FIN")
raw_input()

File: app/DLCosmicTag/tests/run_test.cxx
#include <iostream>
#include <sstream>
#include <string>
#include <vector>

// ROOT
#include "TApplication.h"

// LArCV
#include "Processor/ProcessDriver.h"
#include "Base/LArCVBaseUtilFunc.h"
#include "Base/larcv_logger.h"

#include "DLCosmicTag/DLCosmicTagVertexReco.h"

int main( int nargs, char** argv ) {
  
  std::cout << "This script is used to test the DLCosmicVertexReco" << std::endl;
  
  std::string larcv_input   = argv[1];
  std::string cfg           = "prod_dlcosmictag_vertexreco.cfg";
  
  std::vector<std::string> larcv_input_v;
  larcv_input_v.push_back( larcv_input );

  // auto pset = larcv::CreatePSetFromFile(cfg);
  // auto proccfg = pset.get<larcv::PSet>( "ProcessDriver" );  
  // std::cout << proccfg.dump() << std::endl;

  larcv::ProcessDriver proc("ProcessDriver");  
  proc.configure( cfg );
  proc.override_input_file( larcv_input_v );
  proc.override_output_file( "output_test_dlcosmictagreco.root" );
  proc.override_output_file( "output_test_dlcosmictagreco.root" );  
  proc.initialize();

  int nentries = proc.io().get_n_entries();

  larcv::logger logger = larcv::logger::get_shared();

  std::stringstream msg;
  msg << "Number of events: " << nentries << std::endl;
  logger.send((larcv::msg::Level_t)2,"Setup") << msg.str();

  TApplication app( "app", &nargs, argv );
  
  for ( size_t ientry=0; ientry<nentries; ientry++ ) {
  //for ( size_t ientry=4; ientry<5; ientry++ ) {
    logger.send((larcv::msg::Level_t)2,"EVENT LOOP") << "entry " << ientry << std::endl;
    proc.process_entry(ientry);
  }
  
  logger.send((larcv::msg::Level_t)2,"POST") << "finalize" << std::endl;
  proc.finalize();

  logger.send((larcv::msg::Level_t)2,"POST") << "FIN" << std::endl;  

  return 0;
}
